controller/conditional_request.go

```diff
@@ -0,0 +1,329 @@
+package controller
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/base64"
+	"strconv"
+	"time"
+
+	"github.com/almighty/almighty-core/app"
+)
+
+const (
+	// IfModifiedSince the "If-Modified-Since" HTTP request header name
+	IfModifiedSince = "If-Modified-Since"
+	// LastModified the "Last-Modified" HTTP response header name
+	LastModified = "Last-Modified"
+	// IfNoneMatch the "If-None-Match" HTTP request header name
+	IfNoneMatch = "If-None-Match"
+	// ETag the "ETag" HTTP response header name
+	// should be ETag but GOA will convert it to "Etag" when setting the header.
+	// Plus, RFC 2616 specifies that header names are case insensitive:
+	// https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
+	ETag = "Etag"
+	// CacheControl the "Cache-Control" HTTP response header name
+	CacheControl = "Cache-Control"
+	// MaxAge the "max-age" HTTP response header value
+	MaxAge = "max-age"
+)
+
+// ConditionalRequestContext interface with methods for conditional requests contexts
+type ConditionalRequestContext interface {
+	OK(entity interface{}) error
```

Comment:
This should have been `GenericOK(entity interface{}) error` and I already removed it after your comment above.